### PR TITLE
Include resource and data source name as a comment in user-agent

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

This is to include resource or data source name in the user agent of the API calls.

### Automated Testing

Which runs very looooong and may have intermittent errors.
```bash
make test
```

### Manual Testing

- Enable logging:
```bash
export TF_LOG=DEBUG
export TF_LOG_PATH=$PWD/log.txt 
export TF_LOG_PROVIDER_LINODE_REQUESTS=DEBUG
```
- Deploy some resources, like:
```terraform
resource "linode_vpc" "test" {
    label = "test-vpc${count.index}"
    region = "us-mia"
    description = "My first VPC."
    count = 3
}

resource "linode_nodebalancer" "foobar" {
    label = "mynodebalancer${count.index}"
    region = "us-mia"
    client_conn_throttle = 20
    count = 3
}
```
- Inspect `log.txt` for user agents recorded, they should be something like `User-Agent="HashiCorp Terraform/1.8.1 (+https://www.terraform.io) Terraform-Plugin-Framework/1.8.0 terraform-provider-linode/dev (linode_vpc resource)"`

### CODE REVIEW

File: linode/helper/framework_resource_base.go [hunk 1]
```diff
@@ -55,6 +59,8 @@ func (r *BaseResource) Configure(
 		return
 	}
 
+	r.Client = GetFwClientWithUserAgent(r.Config.Name+" resource", r.Meta)
```
Comment (lgarber-akamai) [inline linode/helper/framework_resource_base.go:62]: Optional nitpick: Would it make sense to distinguish resources/data sources in the UA using the `data.` prefix?

e.g.
`linode_image` vs `data.linode_image`?

File: linode/helper/provider_meta.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	meta *FrameworkProviderMeta,
+) *linodego.Client {
+	client := meta.Client
+	client.SetUserAgent(generate(meta.ProviderUserAgent, resourceOrDataSourceUAComment))
+
+	return &client
+}
+
+func GetSDKClientWithUserAgent(
+	resourceOrDataSourceUAComment string,
```
Comment (lgarber-akamai) [inline linode/helper/provider_meta.go:20]: Just to simplify and unify these helpers a bit, would it make sense for them to accept the resource name and whether it's a resource or data source (enum?) as separate arguments?

File: linode/framework_provider_config.go [hunk 3]
```diff
@@ -290,16 +290,17 @@ func (fp *FrameworkProvider) InitProvider(
...
 	helper.ApplyAllRetryConditions(&client)
 
-	meta.Config = lpm
-	meta.Client = &client
+	fp.Meta.Config = lpm
+	fp.Meta.Client = client
 }
 
-func (fp *FrameworkProvider) terraformUserAgent(
+func (fp *FrameworkProvider) fwProviderUserAgent(
```
Comment (lgarber-akamai) [inline linode/framework_provider_config.go:303]: nitpick: I think this can just become `userAgent(...)` since fwProvider is implied from the struct?

**Review by jriddle-linode — APPROVED** [review]
LGTM, works locally, approved pending Lena's recommendations.

File: linode/helper/provider_meta.go (hunk 2 above)
Comment (zliang-akamai) [reply linode/helper/provider_meta.go:20]: They takes different struct as parameters (our Framework provider metadata and SDKv2 provider metadata are separate structs), so I think it wouldn't look great if we combine them. What do you think?

File: linode/helper/framework_resource_base.go (hunk 1 above)
Comment (zliang-akamai) [reply linode/helper/framework_resource_base.go:62]: Thanks for the suggestion! Just added it.

File: linode/helper/provider_meta.go (hunk 2 above)
Comment (lgarber-akamai) [reply linode/helper/provider_meta.go:20]: Good point, I think it makes sense to leave it as-is in that case 👍

**Review by lgarber-akamai — APPROVED** [review]
LGTM, great work!

**Pull request merged**